R/formatter.r

```diff
@@ -232,32 +232,70 @@ scientific <- function(x, digits = 3, ...) {
 
 #' Ordinal formatter: add ordinal suffixes (-st, -nd, -rd, -th) to numbers.
 #'
-#' @return a function with single parameter x, a numeric vector, that
+#' @return A function with single parameter `x`, a numeric vector, that
 #'   returns a character vector
-#' @param x a numeric vector to format
+#' @param x A numeric vector of positive values to format.
+#' @param prefix,suffix Symbols to display before and after value.
+#' @param big.mark Character used between every 3 digits to separate thousands.
+#' @param rules Custom rules for computing ordinal indicators (list of regular
+#'   expressions, see examples).
+#' @param ... Other arguments passed on to [base::format()].
+#' @note
+#' Values in `x` will be rounded before formating.
 #' @export
 #' @examples
 #' ordinal_format()(1:10)
 #' ordinal(1:10)
-ordinal_format <- function() {
-  function(x) ordinal(x)
+#'
+#'  # Custom rules for French
+#' french <- list(
+#'   er = "^1$",
+#'   nd = "^2$",
+#'   e = "([0-9]+)[12]$",
+#'   e = "[03456789]"
+#' )
+#' ordinal(1:20, rules = french)
+ordinal_format <- function(prefix = "", suffix = "", big.mark = " ",
+                           rules = NULL, ...) {
+  force_all(prefix, suffix, big.mark, rules, ...)
+  function(x) ordinal(
+    x,
+    prefix = prefix,
+    suffix = suffix,
+    big.mark = big.mark,
+    rules = rules,
+    ...
+  )
 }
 
 #' @export
 #' @rdname ordinal_format
-ordinal <- function(x) {
+ordinal <- function(x, prefix = "", suffix = "", big.mark = " ",
+                    rules = NULL, ...) {
   stopifnot(all(x > 0))
+  if (is.null(rules)) {
+    rules <- list(
```

Comment:
Maybe we should make this `ordinal_english()` and export it? Then we can use as the default argument, and users can easily see what the default rules are.

Reply:
done

also added `ordinal_french()` and `ordinal_spanish()`